spec/models/shift_spec.rb

```diff
@@ -0,0 +1,39 @@
+RSpec.describe Shift, type: :model do
+  let(:shift) { FactoryBot.create(:shift) }
+
+  it "is valid with valid attributes" do
+    expect(shift).to be_valid
+  end
+
+  it "is not valid without a user" do
+    shift.user = nil
+    expect(shift).to_not be_valid
+    expect("User must be present").to eq(shift.errors.full_messages.last)
+  end
+
+  it "is not valid without a schedule" do
+    shift.schedule = nil
+    expect(shift).to_not be_valid
+    expect("Schedule must be present").to eq(shift.errors.full_messages.last)
+  end
+
+  it "is not valid without a shift_start" do
+    shift.shift_start = nil
+    expect(shift).to_not be_valid
+    expect("Shift start must be present").to eq(shift.errors.full_messages.last)
+  end
+
+  it "is not valid without a shift_end" do
+    shift.shift_end = nil
+    expect(shift).to_not be_valid
+    expect("Shift end must be present").to eq(shift.errors.full_messages.last)
+  end
+
+  it "returns the formatted shift_start using format_shift" do
+    expect(shift.format_shift(shift.shift_start)).to eq(shift.shift_start.strftime("%b %d, %Y @ %H:%M"))
```

Comment:
So this is what would be referred to as a test that is the implementation.  The test and the implementation are the same code, because the date itself is not fixed.  I would suggest the following:

- Alter the shift creation on L2 to pass in a specific shift start and shift end date
- Hard-code the formatted date in the expectation

This allows your test to have fixed inputs and outputs and more clarity about what the expected output should be.